### PR TITLE
FISH-676 JDBC Config Source Fixes

### PR DESCRIPTION
**Pull request by MattGill98**
Fixes errors found in JDBC config source testing where errors were being swallowed and resources were leaking.

### CODE REVIEW

**Comment by MattGill98** [comment]
jenkins test please

**Comment by MattGill98** [comment]
jenkins test please

File: nucleus/payara-modules/nucleus-microprofile/config-service/src/main/java/fish/payara/nucleus/microprofile/config/source/JDBCConfigSource.java [hunk 1]
```diff
@@ -64,17 +83,31 @@ public int getOrdinal() {
...
 
     @Override
     public String getName() {
         return "JDBC";
     }
 
-    private void init() {
-        if (jdbcConfigHelper == null) {
-            jdbcConfigHelper = new JDBCConfigSourceHelper(Globals.getDefaultHabitat().getService(JDBCConfigSourceConfiguration.class));
+    private JDBCConfigSourceHelper getHelper() {
```
Comment (svendiedrichsen) [inline nucleus/payara-modules/nucleus-microprofile/config-service/src/main/java/fish/payara/nucleus/microprofile/config/source/JDBCConfigSource.java:106]: How about an Optional here?

File: nucleus/payara-modules/nucleus-microprofile/config-service/src/main/java/fish/payara/nucleus/microprofile/config/source/JDBCConfigSource.java [hunk 2]
```diff
@@ -64,17 +83,31 @@ public int getOrdinal() {
 
     @Override
     public String getValue(String propertyName) {
-        return jdbcConfigHelper.getConfigValue(propertyName);
+        JDBCConfigSourceHelper helper = getHelper();
+        if (helper == null) {
+            return null;
+        }
+        try {
+            return helper.getConfigValue(propertyName);
```
Comment (svendiedrichsen) [inline nucleus/payara-modules/nucleus-microprofile/config-service/src/main/java/fish/payara/nucleus/microprofile/config/source/JDBCConfigSource.java:91]: How about try-with-resources here?

File: nucleus/payara-modules/nucleus-microprofile/config-service/src/main/java/fish/payara/nucleus/microprofile/config/source/JDBCConfigSource.java [hunk 3]
```diff
@@ -39,22 +39,41 @@
...
 
     @Override
     public Map<String, String> getProperties() {
-        return jdbcConfigHelper.getAllConfigValues();
+        JDBCConfigSourceHelper helper = getHelper();
+        if (helper == null) {
+            return null;
+        }
+        try {
+            return helper.getAllConfigValues();
```
Comment (svendiedrichsen) [inline nucleus/payara-modules/nucleus-microprofile/config-service/src/main/java/fish/payara/nucleus/microprofile/config/source/JDBCConfigSource.java:69]: How about try-with-resources here?

**Pull request merged**